### PR TITLE
fix or add missed document.

### PR DESCRIPTION
**Pull request by idkiller**
- fix typo.
- remove invalid build precondition
- remove argument from jerry_api_create_object description
- add jerry_api_create_array_object description
- add jerry_api_set_array_index_value description
- add jerry_api_get_array_index_value description
- add jerry_api_release_value description
- add jerry_api_create_error description

JerryScript-DCO-1.0-Signed-off-by: pius.lee pius.lee@samsung.com

### CODE REVIEW

File: 03.api.md [hunk 1]
```diff
@@ -1099,3 +1098,228 @@ handler (const jerry_api_object_t * function_obj_p,
...
+{% highlight cpp %}
+static bool
+handler (const jerry_api_object_t * function_obj_p,
+         const jerry_api_value_t * this_p,
+         jerry_api_value_t * ret_val_p,
+         const jerry_api_value_t args_p[],
+         const uint16_t args_cnt)
+{
+  jerry_api_object_t * error_p = jerry_api_create_error (JERRY_API_ERROR_TYPE,
+                                                           (jerry_api_char_t *) "error");
```
Comment (egavrin) [inline 03.api.md]: Seams that formatting is broken here.

**Comment by egavrin** [comment]
Looks good

File: 03.api.md [hunk 2]
```diff
@@ -1099,3 +1098,228 @@ handler (const jerry_api_object_t * function_obj_p,
 - [jerry_api_is_function](#jerryapiisfunction)
 - [jerry_api_call_function](#jerryapicallfunction)
 - [jerry_api_release_object](#jerryapireleaseobject)
+
+# jerry_api_create_array_object
+
+**Summary**
+Create new Javascript array object.
```
Comment (galpeter) [inline 03.api.md]: Other parts in the document uses the 'JavaScript' casing, we should follow that.

File: 03.api.md [hunk 3]
```diff
@@ -1099,3 +1098,228 @@ handler (const jerry_api_object_t * function_obj_p,
...
+{% endhighlight %}
+
+**See also**
+
+- [jerry_api_value_t](#jerryapivaluet)
+
+# jerry_api_create_error
+
+**Summary**
+Create new Javascript error object.
```
Comment (galpeter) [inline 03.api.md]: Other parts in the document uses the 'JavaScript' casing, we should follow that.

File: 03.api.md [hunk 4]
```diff
@@ -1099,3 +1098,228 @@ handler (const jerry_api_object_t * function_obj_p,
...
+  ret_val_p->type = JERRY_API_DATA_TYPE_OBJECT;
+  ret_val_p->v_object = error_p;
+
+  jerry_api_release_object (error_p);
+
+  return false;
+}
+
+{
+  jerry_api_object_t *throw_obj_p = jerry_api_create_external_function (handler);
```
Comment (galpeter) [inline 03.api.md]: There should be a space after the `*` to match the style used in the file.

File: 03.api.md [hunk 5]
```diff
@@ -1099,3 +1098,228 @@ handler (const jerry_api_object_t * function_obj_p,
...
+- [jerry_api_delete_object_field](#jerryapideleteobjectfield)
+- [jerry_api_get_object_field_value](#jerryapigetobjectfieldvalue)
+- [jerry_api_set_object_field_value](#jerryapisetobjectfieldvalue)
+- [jerry_api_get_object_native_handle](#jerryapigetobjectnativehandle)
+- [jerry_api_set_object_native_handle](#jerryapisetobjectnativehandle)
+
+# jerry_api_set_array_index_value
+
+**Summary**
+Set value of a indexed element in the specified array object.
```
Comment (galpeter) [inline 03.api.md]: typo: .. a indexed .. -> .. a**n** indexed ..

File: 03.api.md [hunk 6]
```diff
@@ -1099,3 +1098,228 @@ handler (const jerry_api_object_t * function_obj_p,
...
+
+**See also**
+
+- [jerry_api_value_t](#jerryapivaluet)
+- [jerry_api_create_array_object](#jerryapicreatearrayobject)
+
+# jerry_api_get_array_index_value
+
+**Summary**
+Get value of a indexed element in the specified array object.
```
Comment (galpeter) [inline 03.api.md]: typo: .. a indexed .. -> .. a**n** indexed ..

**Comment by galpeter** [comment]
looks good to me

**Pull request merged**